shell/platform/darwin/macos/framework/Source/FlutterRenderingBackend.mm

```diff
@@ -4,13 +4,17 @@
 
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterRenderingBackend.h"
 
+#import <Metal/Metal.h>
 #import <QuartzCore/QuartzCore.h>
 
 @implementation FlutterRenderingBackend
 
 + (BOOL)renderUsingMetal {
   if (@available(macOS 10.14, *)) {
-    return YES;
+    // Do not use MTLCreateSystemDefaultDevice() to check if not-null
+    // as that can force a mux switch to the discrete GPU.
```

Comment:
I'm a little confused by this comment. If the only Metal-supporting GPU is the discrete GPU, then presumably this would happen anyway; you don't have any logic here to avoid using Metal if that GPU isn't already active.

Can you elaborate on what the intended behavior is for a multi-GPU device?

Reply:
My intention was to delay the switch from happening until absolutely needed. Which sounds a little premature in retrospect. I will remove change it to be `MTLCreateSystemDefaultDevice()`, this should be fine as that is (default device) indeed the device we use for rendering when there are multiple GPUs.

Reply:
I'm not sure if the following scenario can happen, 2 GPUs one metal capable and the other isn't and `MTLCreateSystemDefaultDevice` returns `nil` as default device isn't capable of metal. This isn't ideal in that scenario, but we will need to address that more broadly.